packages/liveblocks-react/src/room.tsx

```diff
@@ -1075,66 +1073,85 @@ export function createRoomContext<
   }
 
   function incrementQuerySubscribers(queryKey: string) {
-    const requestCache = requestsCache.get(queryKey);
+    const subscribers = subscribersByQuery.get(queryKey);
 
-    if (requestCache === undefined) {
+    if (subscribers === undefined) {
       console.warn(
         `Internal unexpected behavior. Cannot increase subscriber count for query "${queryKey}"`
       );
       return;
     }
 
-    requestCache.subscribers++;
+    subscribersByQuery.set(queryKey, subscribers + 1);
 
     poller.start(POLLING_INTERVAL);
   }
 
   function decrementQuerySubscribers(queryKey: string) {
-    const requestCache = requestsCache.get(queryKey);
+    const subscribers = subscribersByQuery.get(queryKey);
 
-    if (requestCache === undefined || requestCache.subscribers <= 0) {
+    if (subscribers === undefined || subscribers <= 0) {
       console.warn(
         `Internal unexpected behavior. Cannot decrease subscriber count for query "${queryKey}"`
       );
       return;
     }
 
-    requestCache.subscribers--;
+    subscribersByQuery.set(queryKey, subscribers - 1);
 
     let totalSubscribers = 0;
-    for (const requestCache of requestsCache.values()) {
-      totalSubscribers += requestCache.subscribers;
+    for (const subscribers of subscribersByQuery.values()) {
+      totalSubscribers += subscribers;
     }
 
     if (totalSubscribers <= 0) {
       poller.stop();
     }
   }
 
+  /**
+   * Retries an action using the exponential backoff algorithm
+   * @param action The action to retry
+   * @param retryCount The number of times the action has been retried
+   */
+  function handleError(action: () => void, retryCount: number) {
+    if (retryCount >= MAX_ERROR_RETRY_COUNT) return;
+
+    const timeout = Math.pow(2, retryCount) * ERROR_RETRY_INTERVAL;
```

Comment:
This calculates the delay before retrying a failed operation, increasing exponentially with each attempt.

First retry (retryCount = 0):

```ts
timeout = 2^0 * 5000 = 1 * 5000 = 5000ms (5 seconds)
```

Second retry (retryCount = 1):

```ts
timeout = 2^1 * 5000 = 2 * 5000 = 10000ms (10 seconds)
```

and 20000, 40000, 80000 ms and so on.